src/axom/core/AnnotationMacros.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_ANNOTATION_MACROS_HPP_
+#define AXOM_ANNOTATION_MACROS_HPP_
+
+#include "axom/config.hpp"
+#include "axom/core/utilities/Annotations.hpp"
+
+/*!
+ * \def AXOM_ANNOTATE_BEGIN( name )
+ * 
+ * \brief This macro is used to mark the beginning of an annotated region
+ * \note \a AXOM_ANNOTATE_BEGIN must be closed with a matching \a AXOM_ANNOTATE_END
+ */
+#define AXOM_ANNOTATE_BEGIN(name) axom::utilities::annotations::begin(name)
+
+/*!
+ * \def AXOM_ANNOTATE_END( name )
+ * 
+ * \brief This macro is used to mark the end of an annotated region
+ * \note \a AXOM_ANNOTATE_END must close a matching \a AXOM_ANNOTATE_BEGIN
+ */
+#define AXOM_ANNOTATE_END(name) axom::utilities::annotations::end(name)
+
+/*!
+ * \def AXOM_ANNOTATE_SCOPE( name )
+ * \brief This macro is used to annotate a region within an enclosing scope
+ * and is automatically closed when its enclosing scope ends
```

Comment:
I appreciated your explanation in response to my review comment.  Will you please add your words (or something like it) to this Doxygen comment:

"AXOM_ANNOTATE_SCOPE doesn't annotate the portion of the scope before it is called. It adds an RAII variable whose constructor begins an annotation region and whose destructor, which is called at the end of the scope, ends the annotation region."